src/main/scala/fs2/Process1.scala

```diff
@@ -65,15 +65,14 @@ object process1 {
     
   /** Emit the elements of the input `Handle` until the predicate `p` fails, and return the new `Handle`. */      
   def takeWhile[F[_], I](p: I => Boolean)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] = 
-    h => 
-      Pull.await(h) flatMap {
-        case chunk #: h =>
-          chunk.indexWhere(!p(_)) match {
-            case Some(0) => Pull.done
-            case Some(i) => Pull.output(chunk.take(i)) >> Pull.done
-            case None    => Pull.output(chunk) >> takeWhile(p).apply(h)
-          }
-      }
+    Pull.receive {
+      case chunk #: h =>
```

Comment:
For formatting, move the case onto the same line as the receive, unless this would make the line too long.